libcloud/compute/drivers/vcloud.py

```diff
@@ -848,7 +848,8 @@ def _get_auth_token(self):
                      'application/vnd.vmware.vcloud.orgList+xml')).get('href')
             )
 
-            self.connection.set_http_proxy(self.proxy_url)
+            if self.proxy_url:
```

Comment:
I think the coding standards call for strict comparisons to None, like ```if self.proxy_url is not None```.  Other than that, this looks like a no-brainer fix to me.